common/src/chain/config/checkpoints.rs

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::BTreeMap;
+
+use crate::{
+    chain::GenBlock,
+    primitives::{BlockHeight, Id},
+};
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct Checkpoints {
+    checkpoints: BTreeMap<BlockHeight, Id<GenBlock>>,
+}
+
+impl From<BTreeMap<BlockHeight, Id<GenBlock>>> for Checkpoints {
+    fn from(checkpoints: BTreeMap<BlockHeight, Id<GenBlock>>) -> Self {
+        Self::new(checkpoints)
+    }
+}
+
+impl Checkpoints {
+    pub fn new(checkpoints: BTreeMap<BlockHeight, Id<GenBlock>>) -> Self {
+        checkpoints
+            .get(&BlockHeight::new(0))
+            .expect("Checkpoints must have genesis id at height 0");
```

Comment:
Maybe additionally classify the result to prove that it's genesis at height 0?

Reply:
Classify requires ChainConfig... if that were available, we wouldn't need this. Passing ChainConfig here isn't possible because it's a circular dependency. `Checkpoints` is part of ChainConfig.

Reply:
I see, then the message should be changed to "Checkpoints must have a genblock id at height 0" because we can't be sure that it's genesis. 

Also this check can be moved to ChainConfig to validate rules for checkpoints.

Reply:
I don't see the difference with what you suggested in the message... maybe I'm just tired.

Also, checking is circular in chain config. There's no way to do it without setting the value.

Reply:
I mean that check will pass if there will be not genesis block at index 0. But the message suggests otherwise

Reply:
That's something that can only be validated outside. In something like chainstate. Here, it's not possible unless we pass Genesis itself, which can be lied with too.

Reply:
I agree now that's why I just suggest changing the message only